ray-operator/test/e2e/support.go

```diff
@@ -175,3 +179,50 @@ func jobSubmitterPodTemplateApplyConfiguration() *corev1ac.PodTemplateSpecApplyC
 						corev1.ResourceMemory: resource.MustParse("500Mi"),
 					}))))
 }
+
+func redisDeploymentApplyConfigurationWithoutPassword(namespace string) *appsv1ac.DeploymentApplyConfiguration {
```

Comment:
Merge `redisDeploymentApplyConfigurationWithoutPassword`, `redisDeploymentApplyConfiguration`, and `redisServiceApplyConfiguration` into a single utility function because they are related. For example, the labels in the Redis deployment should match the selector for the Kubernetes service.

For example,

```go
func deployRedis(namespace, password) {
  // Create Redis deployment
  // Create K8s service
}
```